source/MaterialXCore/Element.h

```diff
@@ -117,6 +117,13 @@ class Element : public std::enable_shared_from_this<Element>
     ///    the returned path will be relative to this ancestor.
     string getNamePath(ConstElementPtr relativeTo = nullptr) const;
 
+    /// Return the element referred to by the path relative to the current
+    /// element. A path of empty string returns a pointer to the current
+    /// element.
+    /// @param path The path to use to find an element relative to the
```

Comment:
You should describe what happens if the path is the Elements name or if the path is not found. I believe that you also end up returning the Element. So "failure" I assume always menas to return itself?

Reply:
No, if the path is not found, then I believe ElementPtr() is returned.

Reply:
+1, If no exact match if found it would be better to return nullptr.

Reply:
Okay. Still missing a comment if the path if not found what is returned.
Also, I think the English is a bit funny. Maybe something like: "If an empty string is provided as the path then a shared pointer to the current element is returned. If the path cannot be found then an empty shared pointer is returned" ?